apps/methodologies/bold/rule-processors/mass/maximum-distance/src/lib/maximum-distance.processor.ts

```diff
@@ -1,76 +1,71 @@
-import {
-  and,
-  eventNameIsAnyOf,
-  metadataAttributeValueIsAnyOf,
-} from '@carrot-fndn/methodologies/bold/predicates';
+import type { EvaluateResultOutput } from '@carrot-fndn/shared/rule/standard-data-processor';
+
+import { metadataAttributeValueIsAnyOf } from '@carrot-fndn/methodologies/bold/predicates';
+import { ParentDocumentRuleProcessor } from '@carrot-fndn/methodologies/bold/processors';
 import {
+  type Document,
+  type DocumentEvent,
   DocumentEventAttributeName,
   DocumentEventMoveType,
-  DocumentEventName,
 } from '@carrot-fndn/methodologies/bold/types';
-import {
-  DOCUMENT_NOT_FOUND_RESULT_COMMENT,
-  calculateDistanceBetweenTwoEvents,
-} from '@carrot-fndn/methodologies/bold/utils';
-import { RuleDataProcessor } from '@carrot-fndn/shared/app/types';
-import { toDocumentKey } from '@carrot-fndn/shared/helpers';
-import { mapToRuleOutput } from '@carrot-fndn/shared/rule/result';
-import {
-  type RuleInput,
-  type RuleOutput,
-  RuleOutputStatus,
-} from '@carrot-fndn/shared/rule/types';
+import { calculateDistanceBetweenTwoEvents } from '@carrot-fndn/methodologies/bold/utils';
+import { isNil } from '@carrot-fndn/shared/helpers';
+import { RuleOutputStatus } from '@carrot-fndn/shared/rule/types';
 
-import { MAXIMUM_DISTANCE_RESULT_COMMENT } from './maximum-distance.processor.constants';
+const { MOVE_TYPE } = DocumentEventAttributeName;
+const { DROP_OFF, PICK_UP, SHIPMENT_REQUEST } = DocumentEventMoveType;
 
-export class MaximumDistanceProcessor extends RuleDataProcessor {
-  async process(ruleInput: RuleInput): Promise<RuleOutput> {
-    const document = await loadParentDocument(
-      this.context.documentLoaderService,
-      toDocumentKey({
-        documentId: ruleInput.parentDocumentId,
-        documentKeyPrefix: ruleInput.documentKeyPrefix,
-      }),
-    );
+interface Subject {
+  dropOffEvent?: DocumentEvent | undefined;
+  pickUpOrShipmentRequestEvent?: DocumentEvent | undefined;
+}
+
+export class MaximumDistanceProcessor extends ParentDocumentRuleProcessor<Subject> {
+  private ResultComment = {
+    DROP_OFF_NOT_FOUND:
+      'Event with metadata attribute name move-type and value Drop-off was not found',
+    PICK_UP_NOT_FOUND:
+      'Event with metadata attribute name move-type and value Pick-up or Shipment-request was not found',
+  };
 
-    if (!document) {
-      return mapToRuleOutput(ruleInput, RuleOutputStatus.REJECTED, {
-        resultComment: DOCUMENT_NOT_FOUND_RESULT_COMMENT,
-      });
+  protected override evaluateResult({
+    dropOffEvent,
+    pickUpOrShipmentRequestEvent,
+  }: Subject): EvaluateResultOutput {
+    if (isNil(dropOffEvent) || isNil(pickUpOrShipmentRequestEvent)) {
+      return {
+        resultComment: pickUpOrShipmentRequestEvent
+          ? this.ResultComment.DROP_OFF_NOT_FOUND
+          : this.ResultComment.PICK_UP_NOT_FOUND,
+        resultStatus: RuleOutputStatus.REJECTED,
+      };
     }
```

Comment:
uma alternativa pra não precisar fazer uma combinação de condicionais seria criar dois ifs, um pro drop off e outro pro pick-up. Fica mais verboso, porém, mais fácil de ler. Os dois funcionam, aí fica a seu critério
```suggestion
    if (isNil(dropOffEvent)) {
      return {
        resultComment: this.ResultComment.DROP_OFF_NOT_FOUND,
        resultStatus: RuleOutputStatus.REJECTED,
      };
    }
    
    if (isNil(pickUpOrShipmentRequestEvent)) {
      return {
        resultComment: this.ResultComment.PICK_UP_NOT_FOUND,
        resultStatus: RuleOutputStatus.REJECTED,
      };
    }
```